go/vt/discovery/topology_watcher.go

```diff
@@ -375,3 +388,31 @@ func (fbk *FilterByKeyspace) IsIncluded(tablet *topodata.Tablet) bool {
 	_, exist := fbk.keyspaces[tablet.Keyspace]
 	return exist
 }
+
+// FilterByTabletTags is a filter that filters tablets by tablet tag key/values.
+type FilterByTabletTags struct {
+	tags map[string]string
+}
+
+// NewFilterByTabletTags creates a new FilterByTabletTags. All tablets that match
+// all tablet tags will be forwarded to the TopologyWatcher's consumer.
+func NewFilterByTabletTags(tabletTags map[string]string) *FilterByTabletTags {
+	return &FilterByTabletTags{
+		tags: tabletTags,
+	}
+}
+
+// IsIncluded returns true if the tablet's tags match what we expect.
+func (fbtg *FilterByTabletTags) IsIncluded(tablet *topodata.Tablet) bool {
+	if fbtg.tags == nil {
+		return true
+	} else if tablet.Tags == nil {
+		return false
+	}
```

Comment:
code styling: no need for `else`, because you've `return`ed in the `if` statement. The vitess code style prefers early returns and code unindents.

Accept this suggested change at your preference. I will wait until Monday noon/afternoon before merging as is.

```suggestion
	}
	if tablet.Tags == nil {
		return false
	}
```

Reply:
Consistency is key. Sounds good 👍